mxcube3/ho_mediators/beamline_setup.py

```diff
@@ -32,44 +32,80 @@ def __init__(self, beamline_setup):
 
     def getObjectByRole(self, name):
         ho = self._bl.getObjectByRole(name.lower())
```

Comment:
Don't you think the try...except should just be around this line instead of the whole block ? I prefer small try...except blocks, in general.